frontend/src/packages/articles/articles-api.package.ts

```diff
@@ -89,6 +90,39 @@ class ArticleApi extends HttpApi {
     return await response.json<ArticleBaseResponseDto>();
   }
 
+  public async share(id: string): Promise<{ link: string }> {
+    const response = await this.load(
+      this.getFullEndpoint(ArticlesApiPath.$ID_SHARE, { id }),
+      {
+        method: 'GET',
+        contentType: ContentType.JSON,
+        hasAuth: true,
+      },
+    );
+
+    const { link } = await response.json<{ link: string }>();
+
+    await writeTextInClipboard(link);
+
+    return { link };
+  }
+
+  public async getByToken(token: string): Promise<ArticleBaseResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(ArticlesApiPath.SHARED_BASE, {}),
+      {
+        method: 'GET',
+        contentType: ContentType.JSON,
+        hasAuth: false,
+        customHeaders: {
+          [CustomHttpHeader.SHARED_ARTICLE_TOKEN]: token,
+        } as Record<keyof typeof CustomHttpHeader, 'string'>,
```

Comment:
I believe you don't need to use type cast if you have right type and passing right value
```suggestion
        },
```